lib/service/database.js

```diff
@@ -294,7 +298,8 @@ DatabaseService.prototype.queueAnalysisOperations = function(analysis, callback)
                     id: asyncQueryId(analysis, nodeToUpdate),
                     query: transactionQuery([
                         deleteFromCacheTableQuery(targetTableName),
-                        populateCacheTableQuery(targetTableName, nodeToUpdate.sql())
+                        populateCacheTableQuery(targetTableName, nodeToUpdate.sql()),
+                        checkCacheTableQuery(targetTableName)
```

Comment:
I was thinking about this again. If we do the check after the table has been already created we are just making the analysis to fail, but we are still being able to insert into the table. Should we check before trying to insert? So if any previous table went over the limit we don't allow to create any new one.